packages/mix/lib/src/widgets/widget_state/pressable_widget.dart

```diff
@@ -277,15 +279,27 @@ abstract class _MixStateWidgetBuilderState<T extends _MixStateWidgetBuilder>
     setState(fn);
   }
 
-  Future<void> unpressAfterDelay(int initialPressCount) async {
-    final delay = widget.unpressDelay;
+  Timer? _timer;
 
-    if (delay != Duration.zero) {
-      await Future.delayed(delay);
+  @override
+  void dispose() {
+    _timer?.cancel();
+    super.dispose();
+  }
+
+  void unpressAfterDelay(int initialPressCount) {
+    void unpressCallback() {
+      if (_isPressed && _pressCount == initialPressCount) {
+        updateState(() => _isPressed = false);
+      }
     }
 
-    if (_isPressed && _pressCount == initialPressCount) {
-      updateState(() => _isPressed = false);
+    final delay = widget.unpressDelay;
+
+    if (delay != Duration.zero) {
+      _timer = Timer(delay, unpressCallback);
```

Comment:
Look into canceling the timer before setting a new one. Not sure if it applies to this.

Reply:
Done 👍🏻